pkg/apis/components/v1alpha1/types.go

```diff
@@ -20,40 +20,70 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// +genclient
-// +genclient:noStatus
-// +kubebuilder:object:root=true
+//+genclient
+//+genclient:noStatus
+//+kubebuilder:object:root=true
 
 // Component describes an Dapr component type.
 type Component struct {
 	metav1.TypeMeta `json:",inline"`
-	// +optional
+	//+optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	// +optional
+	//+optional
 	Spec ComponentSpec `json:"spec,omitempty"`
-	// +optional
+	//+optional
 	Auth `json:"auth,omitempty"`
-	// +optional
+	//+optional
 	Scopes []string `json:"scopes,omitempty"`
 }
 
+// Kind returns the component kind.
+func (Component) Kind() string {
+	return "Component"
+}
+
+//+genclient
+//+genclient:noStatus
+//+kubebuilder:object:root=true
+
+// PluggableComponent is a especification for a pluggable component.
+type PluggableComponent struct {
+	metav1.TypeMeta `json:",inline"`
+	//+optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	//+optional
+	Scopes []string               `json:"scopes,omitempty"`
+	Spec   PluggableComponentSpec `json:"spec,omitempty"`
+}
+
+// Kind returns the component kind.
+func (PluggableComponent) Kind() string {
+	return "PluggableComponent"
+}
+
+// PluggableComponentSpec is the spec for a pluggable component.
+type PluggableComponentSpec struct {
+	Type    string `json:"type"`
+	Version string `json:"version"`
```

Comment:
Version is not applicable because this only applies to the implementation of the component. In this case, versioning of a pluggable component does not belong to Dapr (redis v1, redis v2, etc). So, we can just remove this attribute.

Reply:
I don't know if I caught your concern,

The version attr is necessary when you have multiples pluggable components versions being used by the same app (i.e making a transition between two majors), it must be used to disambiguate between them. And, currently is being used for the socket name creation - which contains the verison number in it.

So let's say we have the following component

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: state
spec:
  type: state.my-pluggable-component
  version: v1
```

and another one

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: another-state
spec:
  type: state.my-pluggable-component
  version: v2
```

I can't see this as a problem, it's totally fine. So, going back to the problem: We do need two pluggable components definitions to allow such configuration, right? 

They're different applications, different sockets and probably different behaviors,

Please, correct me if I'm wrong

Reply:
OK. I see where you are going. I thought we would not own versioning the pluggable components and let them just be a different type: `state.my-pluggable-component-v2`. But I am OK to keep it as you proposed so we keep consistency with the rest of Dapr.